src/icm20948sensor.cpp

```diff
@@ -0,0 +1,769 @@
+            bias_g[0] = 90;
+            bias_g[1] = 90;
+            bias_g[2] = 90;
+            bias_a[0] = 90;
+            bias_a[1] = 90;
+            bias_a[2] = 90;
+            bias_m[0] = 90;
+            bias_m[1] = 90;
+            bias_m[2] = 90;
+
+            //Sets all bias to 0 in memory
+            imu.SetBiasGyroX(bias_g[0]);
+            imu.SetBiasGyroY(bias_g[1]);
+            imu.SetBiasGyroZ(bias_g[2]);
+
+            imu.SetBiasAccelX(bias_a[0]);
+            imu.SetBiasAccelY(bias_a[1]);
+            imu.SetBiasAccelZ(bias_a[2]);
+
+            imu.SetBiasCPassX(bias_m[0]);
+            imu.SetBiasCPassY(bias_m[1]);
+            imu.SetBiasCPassZ(bias_m[2]);
+
+            //Sets all bias to 0
+            bias_g[0] = 0;
+            bias_g[1] = 0;
+            bias_g[2] = 0;
+            bias_a[0] = 0;
+            bias_a[1] = 0;
+            bias_a[2] = 0;
+            bias_m[0] = 0;
+            bias_m[1] = 0;
+            bias_m[2] = 0;
+
+            //Reloads all bias from memory
+            imu.GetBiasGyroX(&bias_g[0]);
+            imu.GetBiasGyroY(&bias_g[1]);
+            imu.GetBiasGyroZ(&bias_g[2]);
+
+            imu.GetBiasAccelX(&bias_a[0]);
+            imu.GetBiasAccelY(&bias_a[1]);
+            imu.GetBiasAccelZ(&bias_a[2]);
+
+            imu.GetBiasCPassX(&bias_m[0]);
+            imu.GetBiasCPassY(&bias_m[1]);
+            imu.GetBiasCPassZ(&bias_m[2]);
+
+            Serial.println("All set bias should be 90");
+
+            Serial.print("Set Gyro Bias is ");
+            Serial.print(bias_g[0]);
+            Serial.print(",");
+            Serial.print(bias_g[1]);
+            Serial.print(",");
+            Serial.println(bias_g[2]);
+            Serial.print("Set Accel Bias is ");
+            Serial.print(bias_a[0]);
+            Serial.print(",");
+            Serial.print(bias_a[1]);
+            Serial.print(",");
+            Serial.println(bias_a[2]);
+            Serial.print("Set CPass Bias is ");
+            Serial.print(bias_m[0]);
+            Serial.print(",");
+            Serial.print(bias_m[1]);
+            Serial.print(",");
+            Serial.println(bias_m[2]);
+        }
+    #endif
+
+    lastData = millis();
+    working = true;
+
+    #if ESP8266 && defined(SAVE_BIAS)
+        timer.in(bias_save_periods[0] * 1000, [](void *arg) -> bool { ((ICM20948Sensor*)arg)->save_bias(true); return false; }, this);
+    #endif
+}
+
+
+void ICM20948Sensor::motionLoop() {
+    timer.tick();
+    if(imu.dataReady())
+    {
+        lastReset = -1;
+        lastData = millis();
+    }
+    
+
+    if (lastData + 1000 < millis()) {
+        working = false;
+        lastData = millis();        
+        Serial.print("[ERR] Sensor timeout ");
+        Serial.println(addr);
+    }
+}
+
+void ICM20948Sensor::sendData() { 
+    ICM_20948_Status_e readStatus = imu.readDMPdataFromFIFO(&dmpData);
+        if(readStatus == ICM_20948_Stat_Ok || ICM_20948_Stat_FIFOMoreDataAvail)
```

Comment:
This condition will not work as intended. You also need to check whether `readStatus == ICM_20948_Stat_FIFOMoreDataAvail`, otherwise right side is just always true.